# Summary of Scraped Data

In [1]:
import pandas as pd

In [2]:
pharm_df = pd.read_csv("scraped_with_both_synonyms.csv")

In [4]:
contains_bioassay = pharm_df['Link'].str.contains('bioassay', case=False, na=False)

In [7]:
count_bioassay = contains_bioassay.sum()

In [8]:
pharm_df_bioassay = pharm_df[contains_bioassay]

In [10]:
pharm_df_bioassay.describe()

Original Molecule Parsed Molecule  \
count                 20              20   
unique                20              20   
top             BANDAGES        BANDAGES   
freq                   1               1   

                                                    Link Result Type  CAS  \
count                                                 20          20    0   
unique                                                20           1    0   
top     https://pubchem.ncbi.nlm.nih.gov/bioassay/370922    RELEVANT  NaN   
freq                                                   1          20  NaN   

       Deprecated CAS Compound/Substance Synonym? Synonym2?  
count               0                  0       20        20  
unique              0                  0        1         1  
top               NaN                NaN  MISSING   MISSING  
freq              NaN                NaN       20        20

In [13]:
pharm_df_substance = pharm_df[pharm_df["Compound/Substance"] == "SUBSTANCE"]
pharm_df_substance.describe()

Original Molecule    Parsed Molecule  \
count                 323                323   
unique                323                323   
top     ABRUS_PRECATORIUS  ABRUS PRECATORIUS   
freq                    1                  1   

                                                     Link Result Type  \
count                                                 323         323   
unique                                                308           1   
top     https://pubchem.ncbi.nlm.nih.gov/substance/483...    RELEVANT   
freq                                                    5         323   

                CAS                      Deprecated CAS Compound/Substance  \
count            18                                   2                323   
unique           18                                   2                  1   
top     143003-46-7  71892-90-5, 79817-44-0, 87435-86-7          SUBSTANCE   
freq              1                                   1                323   

       Synonym? Synonym2?  
count       323       323  
unique        2         2  
top          NO   MISSING  
freq        277       277

In [24]:
pharm_df_missing = pharm_df[pharm_df['Compound/Substance'].isna()]
pharm_df_missing.describe()

Original Molecule  \
count                                    164   
unique                                   164   
top     2-3-(2-IODOPROPYLIDENEDIOXY)PROPANOL   
freq                                       1   

                             Parsed Molecule  \
count                                    164   
unique                                   164   
top     2,3-(2 IODOPROPYLIDENEDIOXY)PROPANOL   
freq                                       1   

                                            Link Result Type  CAS  \
count                                        115         164    0   
unique                                       114           3    0   
top     https://pubmed.ncbi.nlm.nih.gov/10292123    RELEVANT  NaN   
freq                                           2         106  NaN   

       Deprecated CAS Compound/Substance Synonym? Synonym2?  
count               0                  0      164       164  
unique              0                  0        1         3  
top               NaN                NaN  MISSING   MISSING  
freq              NaN                NaN      164       162

## Summary:
- There are a total of 584 molecules with missing CAS values.
    - Of the 584, 164 are unclassified (Compound v.s. Substance).
    - Another 305 are substances.
    - The remaining 115 are compounds without a CAS.